### ISSUE TITLE
[Enhancement]: Allow the "proceed to checkout" button to be disabled, or other attributes changed

### ISSUE THREAD

**Issue by opr** [woothemes/woocommerce#53751]
### Describe the solution you'd like

In https://github.com/woocommerce/woocommerce/discussions/53615 a user mentioned they want to disable the proceed to checkout button, currently there's no way to do this.

We could do this by allowing the augmentation of button attributes, _or_ we could have a data store entry for tracking the button's enabled/disabled state. My preference is for the former.

### Additional context

CSS Updates are also required for the button as there are no visual cues when it's disabled.

**Comment by pierorocca** [comment]
Reviewed from a product perspective and closing as not planned. 

@yjailin double check please that disabling Place Order button is counter to how checkout is currently designed. We need it enabled to trigger validation errors and there needs to be predictability for developers.

**Comment by yjailin** [comment]
@pierorocca i agree it's the right pattern yes. Let's not disable the button, we need it to trigger error notifications.
